activerecord/lib/active_record/collection_cache_key.rb

```diff
@@ -0,0 +1,27 @@
+module ActiveRecord
+  module CollectionCacheKey
+
+    # Generates a cache key for the records in the given collection.
+    # See <tt>ActiveRecord::Relation#cache_key</tt> for details.
+    def collection_cache_key(collection = all, timestamp_column = :updated_at)
+      query_signature = Digest::MD5.hexdigest(collection.to_sql)
+      key = "#{collection.model_name.cache_key}/query-#{query_signature}"
+
+      size, timestamp = if collection.loaded?
+        [collection.size, collection.collect(&timestamp_column).compact.max]
+      else
+        column_type = collection.klass.type_for_attribute(timestamp_column.to_s)
```

Comment:
In fact, I would even go so far as to say it's worth mentioning `you should not call this method directly` in the documentation.